content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -751,26 +751,154 @@ with the response:
+Xref: [Supported evaluation on CEL](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#evaluation)
+
+Example on rule scoped to the root of a resource with a status subresource:
+```
+{"rule": "self.status.actual <= self.spec.maxDesired"}
+```
+
+- If the Rule is scoped to an object with properties, the accessible properties of the object are field selectable
+  via `self.field` and field presence can be checked via `has(self.field)`. Null valued fields are treated as
+  absent fields in CEL expressions.
+- If the Rule is scoped to an object with additionalProperties (i.e. a map) the value of the map
+  are accessible via `self[mapKey]`, map containment can be checked via `mapKey in self` and all entries of the map
+  are accessible via CEL macros and functions such as `self.all(...)`.
+- If the Rule is scoped to an array, the elements of the array are accessible via `self[i]` and also by macros and
+  functions.
+- If the Rule is scoped to a scalar, `self` is bound to the scalar value.
+  
+Examples:
+
+    - Rule scoped to a map of objects: {"rule": "self.components['Widget'].priority < 10"}
+    - Rule scoped to a list of integers: {"rule": "self.values.all(value, value >= 0 && value < 100)"}
+    - Rule scoped to a string value: {"rule": "self.startsWith('kube')"}
+
+
+The `apiVersion`, `kind`, `metadata.name` and `metadata.generateName` are always accessible from the root of the 
+object and from any x-kubernetes-embedded-resource annotated objects. No other metadata properties are accessible.
+	
+Unknown data preserved in custom resources via `x-kubernetes-preserve-unknown-fields` is not accessible in CEL
+  expressions. This includes:
+  - Unknown field values that are preserved by object schemas with x-kubernetes-preserve-unknown-fields.
+  - Object properties where the property schema is of an "unknown type". An "unknown type" is recursively defined as:
+    - A schema with no type and x-kubernetes-preserve-unknown-fields set to true
+    - An array where the items schema is of an "unknown type"
+    - An object where the additionalProperties schema is of an "unknown type"
+	
+
+Only property names of the form `[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible.
+Accessible property names are escaped according to the following rules when accessed in the expression:
+
+| property name          | escaped to             |
+| -----------------------| -----------------------|
+| `__`                   | `__underscores__`      |
+| `.`                    | `__dot__`              |
+| `-`                    | `__dash__`             |
+| `/`                    | `__slash__`            |
+| [CEL RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax)   | `__{keyword}__`        |
+
+  Examples:
+
+    - Rule accessing a property named "namespace": {"rule": "self.__namespace__ > 0"}
```

Comment:
Maybe switch to a table for these?  I don't really care for the json format being used here in longform docs.

Reply:
Update with a table